File: PDOO/Napakalaki.java/src/napakalakiGame/BadConsequence.java
package napakalakiGame;

import java.util.ArrayList;
/**
 *
 * @author Andres Arco Lopez
 */
public class BadConsequence {
    
    private String text;
    
    private int levels;
    
    private int nVisibleTreasures;
    
    private int nHiddenTreasures;
    
    private boolean death;
    
    private ArrayList<TreasureKind> specificVisibleTreasures = new ArrayList();
    
    private ArrayList<TreasureKind> specificHiddenTreasures = new ArrayList();
    
    public BadConsequence(String text, int levels, int nVisible, int nHidden){
        this.text=text;
        this.levels=levels;
        this.nVisibleTreasures=nVisible;
        this.nHiddenTreasures=nHidden;
        this.death=false;
    }
    
    public BadConsequence(String text, boolean death){
        this.text=text;
        this.levels=0;
        this.nVisibleTreasures=0;
        this.nHiddenTreasures=0;
        this.death=death;
    }
    
    public BadConsequence(String text, int levels, ArrayList<TreasureKind> tVisible, ArrayList<TreasureKind> tHidden){
        this.text=text;
        this.levels=levels;
        this.nVisibleTreasures=0;
        this.nHiddenTreasures=0;
        this.death=death;
        specificVisibleTreasures= tVisible;
        specificHiddenTreasures= tHidden;
    }
    
    public String getText(){
        return text;
    }
    
    public int getLevels(){
        return levels;
    }
    
    public int getnVisibleTreasures(){
        return nVisibleTreasures;
    }
    
    public int getnHiddenTreasures(){
        return nHiddenTreasures;
    }
    
    public boolean getDeath(){
        return death;
    }
    
    public ArrayList<TreasureKind> getSpecificHiddenTreasures(){
        return specificHiddenTreasures;
    }
    
    public ArrayList<TreasureKind> getSpecificVisibleTreasures(){
        return specificVisibleTreasures;
    }
    
    public String toString(){
        String muerte;
        if(death)
            muerte="Si";
        else 
            muerte="No";
        
        return "\n\tTexto: "+ text +"\n\tNiveles perdidos: "+levels+"\n\tMuerte: "
                + muerte + "\n\tTesoros Visbles Perdidos: "+nVisibleTreasures+
                "\n\tTesoros Ocultos Perdidos: "+nHiddenTreasures+ "\n\tTesoros Especificos Visibles Perdidos: "+
                getSpecificVisibleTreasures()+"\n\tTesoros Especificos Ocultos Perdidos: "+  getSpecificHiddenTreasures();
    }
}

File: NapakalakiRuby.ruby/lib/bad_consequence.rb
#encoding: utf-8

module NapakalakiGame
  
  class BadConsequence

    private_class_method :new
     
    MAXTREASURES = 10
    
    private 
    
    def initialize(atext, someLevels, someVisibleTreasures,someHiddenTreasures,someSpecificVisibleTreasures,someSpecificHiddenTreasures,death)
          @text = atext
          @levels = someLevels
          @nVisibleTreasures = someVisibleTreasures
          @nHiddenTreasures = someHiddenTreasures
          @specificVisibleTreasures = someSpecificVisibleTreasures
          @specificHiddenTreasures = someSpecificHiddenTreasures
          @death = death
    end
    
    public
    
    def self.newLevelNumberOfTreasures(aText, someLevels,someVisibleTreasures, someHiddenTreasures)
      new(aText,someLevels,someVisibleTreasures,someHiddenTreasures,[],[],false)
    end

    def self.newLevelSpecificTreasures(aText, someLevels, someSpecificVisibleTreasures, someSpecificHiddenTreasures)
      new(aText,someLevels, 0,0,someSpecificVisibleTreasures,someSpecificHiddenTreasures,false)
    end

    def self.newDeath (aText)
      new(aText,0,0,0,[],[],true)    
    end
    
    def isEmpty
      if @nVisibleTreasures == 0 && @nHiddenTreasures == 0 && @specificVisibleTreasures.empty? && @specificHiddenTreasures.empty?
        empty = true
      else
        empty = false
      end
      return empty
    end
    
    def getLevels
      return @levels      
    end
    
    def getNVisibleTreasures
      return @nVisibleTreasures
    end
    
    def getNHiddenTreasures
      return @nHiddenTreasures
    end
    
    def getSpecificHiddenTreasures
      return @specificHiddenTreasures
    end
    
    def getSpecificVisibleTreasures
      return @specificVisibleTreasures
    end
    
    def substractVisibleTreasures(t)
      #No se sabe todavia
    end
    
    def substractHiddenTreasures(t)
      #No se sabe todavia
    end
    
    def adjustToFitTreasureLists(v, h)
      
    end
    
    def to_s
        mensaje = "\n\tTexto: #{@text}\n\tNiveles perdidos: #{@levels}"   
        mensaje += "\n\tTesoros visibles perdidos: #{@nVisibleTreasures}"
        mensaje += "\n\tTesoros ocultos perdidos: #{@nHiddenTreasures}"
        mensaje += "\n\tTesoros especificos visibles perdidos: #{@specificVisibleTreasures}"
        mensaje += "\n\tTesoros especificos ocultos perdidos: #{@specificHiddenTreasures}"

        if @death
          mensaje += "\n\tMuerte: Si"
        else
          mensaje += "\n\tMuerte: No"
        end
        
        mensaje
    end

  end  
end
